docs/devsecops/DevSecOps.tex

```diff
@@ -0,0 +1,610 @@
+    also put in place a monitoring stack with Prometheus, Grafana and AlertManager which
+    allows us to monitor the health of our services and to be alerted in case of a problem.
+    \medskip \\
+    Not doing that would put us at risk of not knowing if our services are running
+    correctly, and not being alerted in case of a problem.
+\end{itemize}
+
+% Threat model
+\section{Threat model}
+
+We constructed a threat model to identify potential security risks in our project. We 
+created this threat model by starting from the following subject that describes the 
+project:
+
+\medskip
+The project we develop is a platform for seasonal workers to find a job. There's a mobile 
+application for the users that communicate with a microservice backend to get data. The 
+backend get data from multiple databases and an object storage server. The microservices
+are communicating with each other through the NATS message broker. Client authentication 
+is handled by Keycloak. We are running the microservices and the dependencies inside a 
+self-hosted Kubernetes cluster. Our code is stored on GitHub.
+
+\medskip
+To get a better overview of the whole application architecture we used OWASP Threat Dragon
+to make a vulnerabilities diagram
+
+\begin{figure}[H]
+  \centering
+  \includegraphics[width=0.8\textwidth]{imgs/threat_dragon.png}
+  \caption{Threat dragon infrastructure vulnerabilities diagram}
+\end{figure}
+
+\begin{table}[htbp] % Adjusted placement specifier
+  \centering
+  \small % Smaller font size
+  \begin{tabular}{|p{2.5cm}|p{2.5cm}|p{2.5cm}|p{4cm}|p{4cm}|}
+  \hline
+  \textbf{Target} & \textbf{Title} & \textbf{Type} & \textbf{Description} & \textbf{Mitigations} \\ 
+  \hline
+  Mobile app & Man-in-the-middle & Spoofing & A man-in-the-middle attack is a form of cyberattack where an unauthorized third party intercepts and possibly alters communication between two parties without their knowledge or consent. & Since the API uses HTTPS, a man-in-the-middle would only let the attacker get the DNS the user is talking to (if not using a protocol such as DNS over HTTPS) since the HTTPS traffic will be encrypted from end-to-end. \\ 
+  \hline
+  Backend pods & Man-in-the-middle & Privilege escalation & A man-in-the-middle attack is a form of cyberattack where an unauthorized third party intercepts and possibly alters communication between two parties without their knowledge or consent. & Since the API uses HTTPS, and if the machine of the user is not compromised, it is not possible to tamper with the traffic in a useful manner because it is encrypted. \\ 
+  \hline
+  API Gateway & JWT spoofing & Spoofing & JWT spoofing is the deceptive practice of forging or altering JSON Web Tokens to impersonate legitimate users or gain unauthorized access to systems and resources. & Since our authentication token only lasts 5 minutes, we limit the time someone who stole a token can impersonate. \\ 
```

Comment:
I think this is the wrong mitigation explanation for this kind of attack:
```suggestion
  API Gateway & JWT spoofing & Spoofing & JWT spoofing is the deceptive practice of forging or altering JSON Web Tokens to impersonate legitimate users or gain unauthorized access to systems and resources. & Our OIDC access token is signed by Keycloak, and the API gateway checks that the signature is correct on the token provided by the user. Therefore, this kind of attack is not possible. \\ 
```

Reply:
:+1: